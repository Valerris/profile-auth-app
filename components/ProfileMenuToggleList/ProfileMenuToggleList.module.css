.profileMenuToggleList {
	all: unset;
	display: block;
}
