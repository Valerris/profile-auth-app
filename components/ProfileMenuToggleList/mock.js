const mock = [
	{ title: "My Profile", href: "/profile" },
	{ title: "Group Chat", href: "/group-chat" },
	{ title: "Log out", href: "/logout", footer: "true" },
];

export default mock;
