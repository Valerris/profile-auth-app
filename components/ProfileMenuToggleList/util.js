import ProfileMenuToggleItem from "./ProfileMenuToggleItem/ProfileMenuToggleItem";

function createList(list) {
	return list.map((item, i) => (
		<ProfileMenuToggleItem key={i} {...item} />
	));
}

export default { createList };
