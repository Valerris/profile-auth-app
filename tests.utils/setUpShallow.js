export default function (C, props) {
	return shallow(<C {...props} />);
}
