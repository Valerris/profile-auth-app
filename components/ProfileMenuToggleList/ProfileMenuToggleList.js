import { default as React } from "react";
import mock from "./mock";
import util from "./util";
import styles from "./ProfileMenuToggleList.module.css";

function ProfileMenuToggleList({ items }) {
	const UI = util.createList(mock);

	return (
		<ul className={styles.profileMenuToggleList}>{UI ? UI : null}</ul>
	);
}

export default ProfileMenuToggleList;
