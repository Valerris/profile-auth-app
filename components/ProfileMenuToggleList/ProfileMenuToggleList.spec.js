import setUpShallow from "tests.utils/setUpShallow";
import ProfileMenuToggleList from "./ProfileMenuToggleList";

describe("<ProfileMenuToggleList />", () => {
	let cmp = null;

	beforeEach(() => {
		cmp = setUpShallow(ProfileMenuToggleList);
	});

	it("should render ProfileMenuToggleList component", () => {
		expect(cmp).toMatchSnapshot();
	});
});
