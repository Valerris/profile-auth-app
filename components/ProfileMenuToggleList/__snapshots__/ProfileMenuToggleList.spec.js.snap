// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileMenuToggleList /> should render ProfileMenuToggleList component 1`] = `
<ul
  className="profileMenuToggleList"
>
  <ProfileMenuToggleItem
    href="/profile"
    key="0"
    title="My Profile"
  />
  <ProfileMenuToggleItem
    href="/group-chat"
    key="1"
    title="Group Chat"
  />
  <ProfileMenuToggleItem
    footer="true"
    href="/logout"
    key="2"
    title="Log out"
  />
</ul>
`;
